### ISSUE TITLE
Errors with repo: missing path-browserify,hermes realm init error whenever @realm/react Providers are used

### ISSUE THREAD

**Issue by dryprogrammer** [mongodb-university/realm-template-apps#194]
### Errors with repo:
- missing path-browserify (easy fix)
- hermes realm init error whenever @realm/react Providers are used (hard fix)
- warning " > @rneui/themed@4.0.0-rc.8" has incorrect peer dependency "@rneui/base@4.0.0-rc.7"  (easy fix)

### Steps to reproduce:
- followed https://www.mongodb.com/docs/atlas/app-services/tutorial/react-native/
- Ubuntu 22.04.4 LTS
- run ```appservices app create \
  --name MyTutorialApp \
  --template react-native.todo.flex \
  --deployment-model global \
  --environment development```
- yarn, yarn start, yarn android

```
error: Error: Unable to resolve module path-browserify from /home/user/Application-0/frontend/react-native.todo.flex/node_modules/realm/dist/platform/react-native/fs.js: path-browserify could not be found within the project or in these directories:
  node_modules/realm/node_modules
  node_modules
  ../../../../../node_modules
  18 | //
  19 | ////////////////////////////////////////////////////////////////////////////
> 20 | const path_browserify_1 = require("path-browserify");
     |                                    ^
  21 | const file_system_1 = require("../file-system");
  22 | const debug_1 = require("../../debug");
  23 | const binding_1 = require("../binding");
    at ModuleResolver.resolveDependency (/home/user/Application-0/frontend/react-native.todo.flex/node_modules/@react-native-community/cli-plugin-metro/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:139:15)


warning " > @rneui/themed@4.0.0-rc.8" has incorrect peer dependency "@rneui/base@4.0.0-rc.7".



 LOG  Running "SyncTutorial" with {"rootTag":11}
 ERROR  Error: Getting '$$typeof' from binding before it was injected, js engine: hermes 
    at loadModuleImplementation (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:328:14)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:59473:43)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:59473:43)
    at AppContainer (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:59317:36)
    at SyncTutorial(RootComponent) (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:107957:28)
 ERROR  Error: Getting '$$typeof' from binding before it was injected, js engine: hermes 
    at beginWork (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:93271:49)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:59473:43)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:59473:43)
    at AppContainer (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:59317:36)
    at SyncTutorial(RootComponent) (http://localhost:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.synctutorial&modulesOnly=false&runModule=true:107957:28)
 ERROR  TypeError: Cannot read property 'AppProvider' of undefined

This error is located at:
    in AppWrapper
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in SyncTutorial(RootComponent), js engine: hermes
 ERROR  TypeError: Cannot read property 'AppProvider' of undefined

This error is located at:
    in AppWrapper
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in SyncTutorial(RootComponent), js engine: hermes
```

**Comment by krollins-mdb** [comment]
@dryprogrammer, thank you for filing this issue!

I can replicate the problem with Yarn. npm works, so there's something weird happening with the template app and Yarn.

Short term fix: use npm (I know, not ideal)
Long term fix: We'll need to dedicate more time to debugging the Yarn issue. I don't have a good estimate of when that might happen.

**Comment by dryprogrammer** [comment]
wow that really worked. thanks for your quick reply. who would've known yarn would be the culprit

**Comment by krollins-mdb** [comment]
I'm happy to hear that it worked for you! Definitely a weird situation. We appreciate you bringing this to our attention.

**Comment by krollins-mdb** [comment]
@dryprogrammer, the JS SDK just [released v12.6.2](https://github.com/realm/realm-js/releases/tag/v12.6.2), which has a [fix for the `path-browserify` error](https://github.com/realm/realm-js/pull/6524/commits/20023d27a2016321f88c2eac7d559d14156c6354). I can get Yarn to work with JS SDK v12.6.2. Would you mind trying again with that version?

You could either update the JS SDK dependency in your existing project or start a fresh one from the template app, as you outlined in this issue.

**Comment by krollins-mdb** [comment]
I'm closing this issue, as it seems to be resolved. Please reopen if you continue to have problems.
